src/lib/providers/emailPasswordProvider.ts

```diff
@@ -25,11 +25,19 @@ export interface EmailPasswordProviderOptions {
     passwordHashingAlgorithm?: HashingAlgorithm;
     getUserByEmail: (email: string, req: any) => Promise<(User & { password: string }) | undefined>;
     saveNonExistingUser?: (data: UserInfo, hashedPassword: string, req: any) => Promise<User>;
-    sendResetPasswordEmail?: (user: User, req: any) => Promise<void>;
+    sendResetPasswordEmail?: (resetToken: string, user: User, req: any) => Promise<void>;
     /** Validate user before the `saveNonExistingUser` function is called */
     validateUser?: (data: UserInfo, password: string, req: any) => Promise<void>;
     /** Do custom sign-in validation steps eg. check number of login attempts etc. */
     validateSignIn?: (email: string, password: string, req: any) => Promise<void>;
+    /** Generate custom reset password token */
+    getResetPasswordToken?: (user: User) => Promise<string>;
+    saveResetPasswordToken?: (token: string, user: User, req?: any) => Promise<void>;
+    /** Do custom reset password token validation eg. check if the token is expired */
```

Comment:
Please add information when the validation takes place in the reset password process

Reply:
`Do custom reset password token validation before the password is changed (before the `updatePassword` function is called) eg. check if the token is expired`